pkg/resource/cfg_map_client.go

```diff
@@ -0,0 +1,226 @@
+//
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+
+package resource
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/golang/glog"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apilabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+
+	"github.com/intel/crd-reconciler-for-kubernetes/pkg/resource/reify"
+	"github.com/intel/crd-reconciler-for-kubernetes/pkg/states"
+)
+
+type configMapClient struct {
+	globalTemplateValues GlobalTemplateValues
+	restClient           rest.Interface
+	resourcePluralForm   string
+	templateFileName     string
+}
+
+// configMapClient returns a new pod client.
+func NewConfigMapClient(globalTemplateValues GlobalTemplateValues, clientSet *kubernetes.Clientset, templateFileName string) Client {
+	return &configMapClient{
+		globalTemplateValues: globalTemplateValues,
+		restClient:           clientSet.CoreV1().RESTClient(),
+		resourcePluralForm:   "configmaps",
+		templateFileName:     templateFileName,
+	}
+}
+
+func (c *configMapClient) Reify(templateValues interface{}) ([]byte, error) {
```

Comment:
Have you run the linter? I would expect this public method would be missing a comment?

Reply:
Logs from linter:
```
Step 6/8 : RUN make lint
 ---> Running in 11b06adad409
gometalinter --config=lint.json --disable=golint ./pkg/...
Removing intermediate container 11b06adad409
 ---> 87071c7150a3
Step 7/8 : RUN make test
 ---> Running in 96ef98f0ab3d
gometalinter --config=lint.json --disable=golint ./pkg/...
go test -cover -v ./pkg/...
```